Automate documentation website updates

penev92 opened the pull request:
So far, the documentation website relies on manually running the workflow to update it and inputting the correct version. This is now automated on tagging a release or playtest.
The other thing here is that now merges to the `bleed` branch will trigger the workflow, which will try to update the dev version of the documentation website to always keep it up-to-date.

penev92 committed:
Parameterized duplicate code in documentation.yml

penev92 committed:
Automated documentation.yml running on git tag

For release and playtest tags.

penev92 committed:
Automated documentation.yml running on bleed merge

RoosterDragon reviewed (approved):
No complaints about automating this. Will trust that you've got the workflow to run cleanly for this.

Mailaender commented:
Typo in the second commit message. You can also squash them.

penev92 commented:
Fixed the typo and rebased for good measure. I went through the deliberate effort of splitting it into two commits so I don't really want to squish them. I think reviewing then with `?w=1` makes it very easy to see what's going on in each one.

PunkPun reviewed (approved):
Let's give it a spin

PunkPun commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/99cdbda80ccff9f3b5b7f776f25c660198218606...3f07453341c71dae872da37b1bdecb77cf9903cf)

penev92 commented:
Apparently having an environment variable called "VERSION" can have horrible consequences:
https://github.com/OpenRA/OpenRA/actions/runs/6931226699/job/18852517817
The cause is explained here:  https://github.com/dotnet/msbuild/issues/2713
Also I did a derp on the wiki job.
